### DISCUSSION TITLE
Plugin version for next.js 13

### DISCUSSION THREAD

**Discussion by pawelbajgrowicz** [Avansai/next-multilingual#39] [General]
This plugin is awesome, it helped so much in development of i18n in next.js. 
I am really thankful @nbouvrette for your great work.

I'm planning to create next project with next.js 13, but i'm not sure whether this version will be supported by next-multilingual. v13 introduced breaking changes in the project structure and tooling, that's why I understand, your plugin will need a lot of work to keep up with next.js.

Anyway, I would like to ask what the plans are for next.js v13?

**Comment by nbouvrette** [comment]
Thanks @pawelbajgrowicz 

Right now Next.js 13 is supported (using the `pages` directory) - our [site](https://www.avansai.com) currently runs using both the latest Next.js version and `next-multilingual`

If you are planning to use the `app` directory then this is a different story. We'll most likely have to do a full rewrite to support it since they dropped the `i18n` configs completely. Also, this package is using Babel and we'd like to move to SWC to leverage the performance of Rust but as you mentioned it's a lot of work.

Our high-level plan right now is to finish a few core features on our main site (which is also the reason de developed this package) and after that we will need to upgrade to the `app` directory which means also updating this package. I'm not too sure how long everything will take to be honest since I barely touched the `app` directory.

I'm a bit worried about rewriting the Babel plugin into Rust but unless we want to lose all the cool features we built, we will have to do that work, most likely sometime this year.

Of course, we welcome all open-source collaboration as long as it's aligned with our i18n philosophy :)
